File: python/task.py
import time

from support_loader import espp

start = time.time()
def task_func():
    global start
    elapsed = time.time() - start
    print(f"[{elapsed:.3f}] Hello from task")
    time.sleep(.5)
    return False # we don't want to stop the task

task = espp.Task(
    task_func,
    espp.Task.BaseConfig("test task")
)
task.start()

time.sleep(5)

exit()

File: pc/tests/task.cpp
#include "task.hpp"

using namespace std::chrono_literals;

int main() {

  static auto start = std::chrono::steady_clock::now();
  static auto elapsed = []() -> float {
    return std::chrono::duration<float>(std::chrono::steady_clock::now() - start).count();
  };

  espp::Logger logger({.tag = "Task Test", .level = espp::Logger::Verbosity::INFO});

  logger.info("Starting task test");

  espp::Task task({.name = "Task", .callback = [&](auto &, auto &) -> bool {
                     auto now = std::chrono::steady_clock::now();
                     logger.info("[{:.3f}] Hello from the task!", elapsed());
                     std::this_thread::sleep_until(now + 1s);
                     // don't want to stop the task
                     return false;
                   }});
  task.start();

  std::this_thread::sleep_for(10s);

  logger.info("Stopping task test");

  return 0;
}
